crates/bevy_pbr/src/render/mesh.rs

```diff
@@ -116,79 +116,43 @@ bitflags::bitflags! {
 
 pub fn extract_meshes(
     mut commands: Commands,
-    mut previous_caster_len: Local<usize>,
-    mut previous_not_caster_len: Local<usize>,
-    caster_query: Query<
-        (
-            Entity,
-            &ComputedVisibility,
-            &GlobalTransform,
-            &Handle<Mesh>,
-            Option<&NotShadowReceiver>,
-        ),
-        Without<NotShadowCaster>,
-    >,
-    not_caster_query: Query<
-        (
-            Entity,
-            &ComputedVisibility,
-            &GlobalTransform,
-            &Handle<Mesh>,
-            Option<&NotShadowReceiver>,
-        ),
-        With<NotShadowCaster>,
-    >,
+    mut prev_len_shadow_caster: Local<usize>,
+    mut prev_len_not_shadow_caster: Local<usize>,
+    meshes_query: Query<(
+        Entity,
+        &ComputedVisibility,
+        &GlobalTransform,
+        &Handle<Mesh>,
+        Option<Without<NotShadowReceiver>>,
```

Comment:
If the component exists, it returns `Some(component)`. Otherwise it returns `None`. 

The particularly tricky bit here is that `Without` is being used in the fetch parameter of the query, and so you're getting a bool returned there.

Reply:
So if it matches `Without<NotShadowCaster>` then it returns `Some(true)` and otherwise `None`?

Reply:
Why not just `Option<&NotShadowCaster>` and invert the logic?

Reply:
I agree with that suggestion; I think this is probably clearer.

Reply:
A few things:
1. Theoretically, `Option<&NotShadowCaster>` requires building a reference based on an offset and stuff, while `Option<With<NotShadowCaster>>` does not, it might have different performance characteristic.
2. Minor, but the item type for `Option<With<NotShadowCaster>>` is `Option<()>` not `Option<bool>`
3. I preferred `Without<Not…>` over `With<Not…>` because in the first implementation I got very confused by the negative on the variables.